rules/CodeQuality/Rector/Attribute/OrderByKeyToClassConstRector.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Doctrine\CodeQuality\Rector\Attribute;
+
+use Doctrine\Common\Collections\Criteria;
+use PhpParser\Node;
+use PhpParser\Node\Attribute;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Scalar\String_;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Doctrine\Tests\CodeQuality\Rector\Attribute\OrderByKeyToClassConstRector\OrderByKeyToClassConstRectorTest
+ */
+final class OrderByKeyToClassConstRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace OrderBy Attribute ASC/DESC with class constant from Criteria',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+use Doctrine\ORM\Mapping as ORM;
+
+class ReplaceOrderByAscWithClassConstant
+{
+    #[ORM\OrderBy(['createdAt' => 'ASC'])]
+    protected \DateTimeInterface $messages;
+}
+?>
+CODE_SAMPLE
+
+                    ,
+                    <<<'CODE_SAMPLE'
+use Doctrine\ORM\Mapping as ORM;
+
+class ReplaceOrderByAscWithClassConstant
+{
+    #[ORM\OrderBy(['createdAt' => \Doctrine\Common\Collections\Criteria::ASC])]
+    protected \DateTimeInterface $messages;
+}
+?>
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Attribute::class];
```

Comment:
is the usage only on Property ?, better to use `Property` node.

Reply:
I think the only usage is on the Attribute `#[ORM\OrderBy]`

Reply:
what I mean is it only on property? then use `PhpParser\Node\Stmt\Property` node instead, and resolve its Attribute from attrGroups values..

Reply:
Yes its only used there, and then use Property because its more efficient? I implemented it

Reply:
Practically yes, because you don't check every Attribute found in any stmt